zipline/testing/fixtures.py

```diff
@@ -108,30 +115,136 @@ def enter_instance_context(self, context_manager):
     def add_instance_callback(self, callback):
         """
         Register a callback to be executed during tearDown.
+
+        Parameters
+        ----------
+        callback : callable
+            The callback to invoke at the end of each test.
         """
         return self._instance_teardown_stack.callback(callback)
 
 
+class WithLogger(object):
+    """ZiplineTestCase mixing providing cls.log_handler as an instance-level
+    fixture.
+
+    after init_instance_fixtures has been called `self.log_handler` will be a
+    new `FileHandler` which has been pushed with the value of
+    `self.LOGGER_PATH`
+    """
+    LOGGER_PATH = 'test.log'
+
+    def init_instance_fixtures(self):
```

Comment:
Is there a reason we're making a new handler and entering it for every test method?